### PR TITLE
Add inline namespace for ABI version

### PR DESCRIPTION
**Pull request by rnburn**

### CODE REVIEW

File: api/include/opentelemetry/version.h [hunk 1]
```diff
@@ -1,4 +1,15 @@
...
 
+#include "opentelemetry/detail/preprocessor.h"
+
+#define OPENTELEMETRY_ABI_VERSION_NO 0
 #define OPENTELEMETRY_VERSION "0.0.1"
-#define OPENTELEMETRY_ABI_VERSION "0"
+#define OPENTELEMETRY_ABI_VERSION OPENTELEMETRY_STRINGIFY(OPENTELEMETRY_ABI_VERSION_NO)
+
+// clang-format off
+#define OPENTELEMETRY_BEGIN_NAMESPACE \
```
Comment (reyang) [inline api/include/opentelemetry/version.h]: Would you consider syntax sugar like `OPENTELEMETRY_BEGIN_NAMESPACE(trace)`?

Comment (rnburn) [reply api/include/opentelemetry/version.h]: There might be more than one nested namesapace (e.g. opentelemetry::v1::sdk::trace).

I tried to make this match what [libc++](https://github.com/llvm-mirror/libcxx) does with their _LIBCPP_BEGIN_NAMESPACE_STD macro

Comment (reyang) [reply api/include/opentelemetry/version.h]: Your approach here is the same of what Microsoft did in the internal C++ SDK.
Regarding nested namespaces, I guess I've seen some macro meta-programming solution which allows things like `OPENTELEMETRY_BEGIN_NAMESPACE(v1, sdk, trace)`.
Just my curiosity, I've already approved the PR so feel free to decide, either case would work for me.

Comment (g-easy) [reply api/include/opentelemetry/version.h]: I like the simple (i.e. current) approach.

**Pull request merged**